src/libpmemset/pmemset.c

```diff
@@ -254,9 +262,15 @@ pmemset_delete(struct pmemset **set)
 		return 0;
 
 	int ret = 0;
+	struct pmemset_config *cfg = pmemset_get_pmemset_config(*set);
+	struct pmem2_vm_reservation *rsv = pmemset_config_get_reservation(cfg);
+
+	struct pmemset_delete_all_pmaps_ravl_cb_arg arg = { .ret = &ret,
+			.adjust_reservation = (rsv == NULL) };
+
 	/* delete RAVL tree with part_map nodes */
 	ravl_interval_delete_cb((*set)->shared_state.part_map_tree,
```

Comment:
you could just make this function pass through the return value of the callback, eliminating the need for this oddly named arg structure.